src/improvements/template/DeputyPauseRotationKey.sol

```diff
@@ -0,0 +1,89 @@
+        string[] memory storageWrites = new string[](2);
+        storageWrites[0] = "DeputyPauseModule";
+        storageWrites[1] = safeAddressString();
+        return storageWrites;
+    }
+
+    /// @notice Sets up the template with implementation configurations from a TOML file.
+    function _templateSetup(string memory taskConfigFilePath) internal override {
+        super._templateSetup(taskConfigFilePath);
+
+        string memory file = vm.readFile(taskConfigFilePath);
+
+        // Load the inputs from the task config
+        new_deputy = vm.parseTomlAddress(file, ".newDeputy");
+        new_deputy_signature = vm.parseTomlBytes(file, ".newDeputySignature");
+    }
+
+    /// @notice Write the calls that you want to execute for the task.
+    function _build() internal override {
+        // Load the DeputyPauseModule contract.
+        IDeputyPauseModule dpm = IDeputyPauseModule(superchainAddrRegistry.get("DeputyPauseModule"));
+        // 1. In the future, check that the DeputyPauseModule address is the one that is enabled in the foundation safe.
+        // assertEq(fos.enabledmodule(address(dpm)), true, "ERR100: DeputyPauseModule is should be enabled");
+        // 2. Check that the input are valid.
+        assertNotEq(dpm.deputy(), address(0), "ERR101: DeputyPauseModule should have already a deputy set in the past.");
+        assertNotEq(dpm.deputy(), new_deputy, "ERR102: DeputyPauseModule should have a new deputy");
+
+        // Check that the rotation signature is valid
+        // The signature should be created by the new_deputy address over their own address
+        // using EIP-712 typed data with the message format "DeputyAuthMessage(address deputy)"
+
+        // Get the typehash for deputy auth messages
+        // bytes32 deputyAuthMessageTypehash = dpm.deputyAuthMessageTypehash();
+
+        // We can't directly call the internal _hashTypedDataV4 function from the contract
+        // but we can validateggj§ the signature is right by checking if it reverts when passed to setDeputy
+        // This serves as a pre-flight check before actual execution
+
+        // Verify the new_deputy_signature is exactly 65 bytes (r, s, v format)
+        require(new_deputy_signature.length == 65, "ERR104: Invalid signature length");
+        // 3. Rotate the new deputy and signature.
+        dpm.setDeputy(new_deputy, new_deputy_signature);
+    }
+
+    /// @notice This method performs all validations and assertions that verify the calls executed as expected.
+    function _validate(VmSafe.AccountAccess[] memory, Action[] memory) internal view override {
+        IDeputyPauseModule dpm = IDeputyPauseModule(superchainAddrRegistry.get("DeputyPauseModule"));
+        assertEq(dpm.deputy(), new_deputy, "ERR103: DeputyPauseModule should have a the new deputy set");
+        // check the foundation has the DPM enabled.
+        // assertEq(fos.enabledmodule(address(dpm)), true, "ERR100: DeputyPauseModule is should be enabled");
```

Comment:
we should read the list of modules from the safe and make sure it's found there

Reply:
This check is probably uncommented because we are rotating before the FND has even installed

Reply:
Correct! I will resolve.